webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/changeOwner/BulkChangeOwnerModal.js

```diff
@@ -0,0 +1,265 @@
+  };
+
+  const handleConfirm = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      owner_id: ownerId,
+    };
+
+    dispatch(bulkChangeOwner(requestBody, handleSuccess, handleError));
+  };
+
+  const modalActions = [
+    <Button
+      key="add"
+      ouiaId="bulk-change-owner-modal-add-button"
+      variant="primary"
+      onClick={handleConfirm}
+      isDisabled={ownerId === ''}
+      isLoading={
+        userStatus === STATUS.PENDING || usergroupStatus === STATUS.PENDING
+      }
+    >
+      {__('Confirm')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-change-owner-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change Owner')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-change-owner-modal"
+      key="bulk-change-owner-modal"
+      ouiaId="bulk-change-owner-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-change-owner-options">
+          {__(
+            'Changing owner will affect all your selected systems. Some systems may already been associated with your chosen owner.'
+          )}
+        </Text>
+      </TextContent>
+      {userStatus === STATUS.RESOLVED && usergroupStatus === STATUS.RESOLVED && (
+        <Select
+          id="single-grouped-select"
+          isOpen={ownerSelectOpen}
+          selected={ownerId}
+          onSelect={handleOwnerSelect}
+          onOpenChange={isSelectOpen => setOwnerSelectOpen(isSelectOpen)}
+          toggle={toggle}
+          shouldFocusToggleOnSelect
+          ouiaId="bulk-change-owner-select"
+        >
+          {users && (
+            <SelectGroup label="Users">
+              <SelectList>
+                {users.results?.map(u => (
+                  <SelectOption key={`${u.id}-Users`} value={`${u.id}-Users`}>
+                    {u.login}
+                  </SelectOption>
+                ))}
+              </SelectList>
+            </SelectGroup>
+          )}
+          {usergroups && (
+            <SelectGroup label="Usergroups">
+              <SelectList>
+                {usergroups?.results?.map(ug => (
+                  <SelectOption
+                    key={`${ug.id}-Usergroups`}
+                    value={`${ug.id}-Usergroups`}
+                  >
+                    {ug.name}
+                  </SelectOption>
+                ))}
+              </SelectList>
+            </SelectGroup>
+          )}
+        </Select>
+      )}
+      <div
+        style={{
+          width: '70%',
+          maxHeight: '50%',
+          marginLeft: '-2rem',
+          marginTop: '1rem',
+        }}
+      >
+        <TreeView
+          data={selectedTreeViewData}
+          aria-label={__('Selected hosts')}
```

Comment:
If the TreeView isn't going to be expandable, I feel like it's better to just have the number of selected hosts as part of the text instead. (Use a FormattedMessage so that the number can be bold and it can still translate properly.)

Reply:
Fixed